source/tutorials/cross-compilation.rst

```diff
@@ -0,0 +1,257 @@
+  pkgsCross.gnu64                       pkgsCross.riscv64-embedded
+  pkgsCross.i686-embedded               pkgsCross.scaleway-c1
+  pkgsCross.iphone32                    pkgsCross.sheevaplug
+  pkgsCross.iphone32-simulator          pkgsCross.vc4
+  pkgsCross.iphone64                    pkgsCross.wasi32
+  pkgsCross.iphone64-simulator          pkgsCross.x86_64-embedded
+  pkgsCross.mingw32                     pkgsCross.x86_64-netbsd
+  pkgsCross.mingwW64                    pkgsCross.x86_64-netbsd-llvm
+  pkgsCross.mmix                        pkgsCross.x86_64-unknown-redox
+  pkgsCross.msp430                      
+
+
+From the attribute name it can't always be immediately clear what is the platform.
+
+It's possible to query the platform name using::
+
+  $ nix-instantiate '<nixpkgs>' -A pkgsCross.aarch64-darwin.hostPlatform.config --eval
+  "aarch64-apple-darwin"
+
+.. note:: In case the plaforms hasn't been defined yet, feel free to contribute one
+          by `adding it upstream <https://github.com/NixOS/nixpkgs/blob/master/lib/systems/examples.nix>`_.
+
+
+Cross-compiling for the first time!
+-----------------------------------
+
+To cross-compile a package like `hello <https://www.gnu.org/software/hello/>`_,
+pick the platform target name like ``aarch64-multiplatform`` in our case and run:
+
+.. code:: shell-session 
+
+  $ nix-build '<nixpkgs>' -A pkgsCross.aarch64-multiplatform.hello
+  ...
+  /nix/store/pzi2h0d60nb4ydcl3nn7cbxxdnibw3sy-hello-aarch64-unknown-linux-gnu-2.10
+
+`Search for a package <https://search.nixos.org/packages>`_ attribute name to find the
+one that you're interested in to build.
+
+
+Real world cross-compiling of a Hello World example
+---------------------------------------------------
+ 
+To show off the power of cross-compilation in Nix, let's build our own Hello World program 
+by cross-compiling it as static executables to ``armv6l-unknown-linux-gnueabihf``
+and ``x86_64-w64-mingw32`` (Windows) platforms and run the resulting executable
+with `an emulator <https://en.wikipedia.org/wiki/Emulator>`_.
+
+.. code:: nix 
+
+  { pkgs ? import <nixpkgs> {} 
+  }:
+
+  let
+    # Create a C program that prints Hello World
+    helloWorld = pkgs.writeText "hello.c" ''
+      #include <stdio.h>
+      
+      int main (void)
+      {
+        printf ("Hello, world!\n");
+        return 0;
+      }
+    '';
+
+    # A function that takes host platform packages
+    crossCompileFor = hostPkgs:
+      # Run a simple command with the compiler available
+      hostPkgs.runCommandCC "hello-world-cross-test" {} ''
+        # Wine requires home directory
+        HOME=$PWD
+
+        # Compile our example using the compiler specific to our host platform
+        $CC ${helloWorld} -o hello 
+
+        # Run the compiled program using an emulator
+        # Usually Qemu, but on windows it is Wine
+        ${hostPkgs.stdenv.hostPlatform.emulator hostPkgs.buildPackages} hello > $out
```

Comment:
Especially `buildPackages` and maybe how the emulator attribute works should be explained.